data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/LogstashPluginMapper.java

```diff
@@ -12,5 +14,5 @@
      * @param logstashPlugin A Logstash plugin with its attributes
      * @return A Data Prepper plugin with its attributes
      */
-    PluginModel mapPlugin(LogstashPlugin logstashPlugin);
+    PluginModel mapPlugin(LogstashPlugin logstashPlugin) throws IOException;
```

Comment:
This should not throw `IOException`. We should wrap any such exceptions in the `LogstashMappingException`.